fv3core/pace/fv3core/stencils/fvtp2d.py

```diff
@@ -336,15 +339,61 @@ def __call__(
             self._q_advected_x, cry, self._q_advected_x_y_advected_mean
         )
 
-        self.stencil_transport_flux(
-            self._q_advected_y_x_advected_mean,
-            self._q_x_advected_mean,
-            self._q_advected_x_y_advected_mean,
-            self._q_y_advected_mean,
-            x_unit_flux,
-            y_unit_flux,
-            q_x_flux,
-            q_y_flux,
-        )
+        # TODO [DACE]: due to an aliiasing issue (see above for original code)
+        # we duplicate the code here
+        if x_mass_flux is None:
+            # x_unit_flux = x_area_flux
+            if y_mass_flux is None:
+                # y_unit_flux = y_area_flux
+                self.stencil_transport_flux(
+                    self._q_advected_y_x_advected_mean,
+                    self._q_x_advected_mean,
+                    self._q_advected_x_y_advected_mean,
+                    self._q_y_advected_mean,
+                    x_area_flux,
+                    y_area_flux,
+                    q_x_flux,
+                    q_y_flux,
+                )
+            else:
+                # y_unit_flux = y_mass_flux
+                self.stencil_transport_flux(
+                    self._q_advected_y_x_advected_mean,
+                    self._q_x_advected_mean,
+                    self._q_advected_x_y_advected_mean,
+                    self._q_y_advected_mean,
+                    x_area_flux,
+                    y_mass_flux,
+                    q_x_flux,
+                    q_y_flux,
+                )
+
+        else:
+            # x_unit_flux = x_mass_flux
+            if y_mass_flux is None:
+                # y_unit_flux = y_area_flux
+                self.stencil_transport_flux(
+                    self._q_advected_y_x_advected_mean,
+                    self._q_x_advected_mean,
+                    self._q_advected_x_y_advected_mean,
+                    self._q_y_advected_mean,
+                    x_mass_flux,
+                    y_area_flux,
+                    q_x_flux,
+                    q_y_flux,
+                )
+            else:
+                # y_unit_flux = y_mass_flux
+                self.stencil_transport_flux(
+                    self._q_advected_y_x_advected_mean,
+                    self._q_x_advected_mean,
+                    self._q_advected_x_y_advected_mean,
+                    self._q_y_advected_mean,
+                    x_mass_flux,
+                    y_mass_flux,
+                    q_x_flux,
+                    q_y_flux,
+                )
+
```

Comment:
Well, we can alias the `x_mass_flux` & `y_mass_flux`, so that really gives us little leeway. I can fold things into a function but it'll be only anecdotally better really. What you want to do is alias

Reply:
Okay here's a slightly better way to write it

Reply:
I guess that does look better, thanks